### DISCUSSION TITLE
Filters MQTT

### DISCUSSION THREAD

**Discussion by mapchen** [syssi/esphome-jk-bms#438] [Q&A] [answered]
Dears,

I got the tool running on one of the new inverter JK BMS. Thanks for that. I have seen that it creates quite some traffic on MQTT. After reading how to change that I came across the FILTERS and HEARTBEAT option in ESPhome.

Tried to put it into the YAML but failed miserably:

# Example filters:
filters:
   - heartbeat: 5s

I have placed it at the end of the sensors section like this:
.....................
  power_tube_temperature:
      name: "${name} power tube temperature"
    state_of_charge:
      name: "${name} state of charge"
    capacity_remaining:
      name: "${name} capacity remaining"
    total_battery_capacity_setting:
      name: "${name} total battery capacity setting"
    charging_cycles:
      name: "${name} charging cycles"
    total_charging_cycle_capacity:
      name: "${name} total charging cycle capacity"
    total_runtime:
      name: "${name} total runtime"
    balancing_current:
      name: "${name} balancing current"
    errors_bitmask:
      name: "${name} errors bitmask"
    filters:
        - heartbeat: 30s
..............

Its telling me:
    name: jk-bms balancing current
  errors_bitmask:
    name: jk-bms errors bitmask

  [filters] is an invalid option for [sensor.jk_bms_ble]. Please check the indentation.
  filters:
    - heartbeat: 30s

Any idea whats wrong here?

**Comment by syssi** [comment]
First of all: Could you explain what do you try to achieve? It looks like the heartbeat filtern isn't the correct approach if you want to reduce the amount of MQTT messages.

**Reply by mapchen** [reply]
thank your for your response.

I would like to have the node publish to MQTT every 30 seconds the sensor values. Now it seems to be much more frequent.

**Reply by syssi** [reply, marked as answer]
Just change the `throttle` parameter of the `jk_bms_ble` section to `30s`.
